### PR TITLE
Improve test coverage for proxy package

### PR DESCRIPTION
**Pull request by n0npax**
Improve test coverage in `proxy` package.
Change `ExcludeIP` to return an error if the passed argument is not `IP` or `CIDR`.

### CODE REVIEW

File: pkg/minikube/proxy/proxy_test.go [hunk 1]
```diff
@@ -55,6 +58,7 @@ func TestIsInBlock(t *testing.T) {
 		{"192.168.0.2", "192.168.0.1/32", false, false},
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/proxy/proxy_test.go]: subtests are printing now `Want %s WantAErr %t`. Hope it's ok.

File: pkg/minikube/proxy/proxy_test.go [hunk 2]
```diff
@@ -135,17 +138,119 @@ func TestCheckEnv(t *testing.T) {
...
+		{"1.2.3.4", "7.7.7.7", false},
+		{"1.2.3.4", "1.2.3.4", true},
+		{"1.2.3.4", "", false},
+		{"foo", "", false},
+		{"foo", "bar", false},
+		{"foo", "1.2.3.4", false},
+	}
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("exclude %s NO_PROXY(%v)", tc.ip, tc.env), func(t *testing.T) {
+			originalEnv := os.Getenv("NO_PROXY")
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/proxy/proxy_test.go]: good hint. Thanks. Fixed

File: pkg/minikube/proxy/proxy_test.go [hunk 3]
```diff
@@ -135,17 +138,119 @@ func TestCheckEnv(t *testing.T) {
...
+		{"1.2.3.4", "", false},
+		{"", "", true},
+		{"7.7.7.7", "7.7.7.7", false},
+		{"7.7.7.7", "1.2.3.4", false},
+		{"foo", "", true},
+		{"foo", "1.2.3.4", true},
+	}
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("exclude %s NO_PROXY(%s)", tc.ip, tc.env), func(t *testing.T) {
+			originalEnv := os.Getenv("NO_PROXY")
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/proxy/proxy_test.go]: fixed! Thanks

File: pkg/minikube/proxy/proxy.go [hunk 4]
```diff
@@ -57,6 +57,11 @@ func isInBlock(ip string, block string) (bool, error) {
 
 // ExcludeIP will exclude the ip from the http(s)_proxy
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/proxy/proxy.go]: Comment improved. I'm not sure if it's good enough, but for sure it's better.

File: pkg/minikube/proxy/proxy.go [hunk 5]
```diff
@@ -55,8 +55,13 @@ func isInBlock(ip string, block string) (bool, error) {
 	return false, errors.Wrapf(err, "Error ip not in block")
 }
 
-// ExcludeIP will exclude the ip from the http(s)_proxy
+// ExcludeIP takes ip or CIDR as string and excludes the it from the http(s)_proxy
```
[the remark this answers is not in the document]
Comment (n0npax) [reply pkg/minikube/proxy/proxy.go]: yes, typo. Fixed.

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by medyagh** [comment]
Haven't checked the integration failure logs. Otherwise looks good

**Comment by josedonizetti** [comment]
@minikube-bot Ok to test

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by n0npax** [comment]
@RA489 could you please approve to tests using `/ok-to-test` syntax? Seems I cannot run retest on my own.

**Pull request merged**